lib/Map/ImageryProvider/ProtomapsImageryProvider.ts

```diff
@@ -248,50 +232,52 @@ export default class ProtomapsImageryProvider
     return [];
   }
 
-  async requestImage(x: number, y: number, level: number) {
+  async requestImage(x: number, y: number, level: number, request: Request) {
     const canvas = document.createElement("canvas");
     canvas.width = this.tileWidth;
     canvas.height = this.tileHeight;
-    return await this.requestImageForCanvas(x, y, level, canvas);
+    return await this.requestImageForCanvas(x, y, level, canvas, request);
   }
 
   async requestImageForCanvas(
     x: number,
     y: number,
     level: number,
-    canvas: HTMLCanvasElement
+    canvas: HTMLCanvasElement,
+    request?: Request | undefined
   ) {
-    try {
-      await this.renderTile({ x, y, z: level }, canvas);
-    } catch (e) {
-      console.log(e);
-    }
+    if (this.softMinimumLevel && level < this.softMinimumLevel)
+      throw TerriaError.from(`Level: ${level} is below softMinimumLevel`);
 
-    return canvas;
-  }
+    const coords: Coords = { z: level, x, y };
 
-  public async renderTile(coords: Coords, canvas: HTMLCanvasElement) {
     // Adapted from https://github.com/protomaps/protomaps.js/blob/master/src/frontends/leaflet.ts
     let tile: PreparedTile | undefined = undefined;
 
     // Get PreparedTile from source or view
     // Here we need a little bit of extra logic for the ProtomapsGeojsonSource
-    if (this.source instanceof ProtomapsGeojsonSource) {
-      const data = await this.source.get(coords, this.tileHeight);
+    if (
+      this.source instanceof ProtomapsGeojsonSource ||
+      this.source instanceof ProtomapsArcGisPbfSource
+    ) {
+      const data = await this.source.get(coords, this.tileHeight, request);
 
       tile = {
         data: data,
         z: coords.z,
         dataTile: coords,
         scale: 1,
-        origin: new Point(coords.x * 256, coords.y * 256),
+        origin: new Point(
+          coords.x * PROTOMAPS_DEFAULT_TILE_SIZE,
+          coords.y * PROTOMAPS_DEFAULT_TILE_SIZE
+        ),
         dim: this.tileWidth
       };
     } else if (this.view) {
       tile = await this.view.getDisplayTile(coords);
     }
 
-    if (!tile) return;
+    if (!tile) throw TerriaError.from("Failed to get tile");
```

Comment:
Is there a tile name that could be included, or is it obvious what tile that failed when this happens?

Reply:
It tile is `undefined` then there are bigger problems - there is no appropriate source, or the `view` hasn't been setup. I have modified the message to reflect this